src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -0,0 +1,312 @@
+    }
+
+    public static int deleteDevice_Transaction(Start start, Connection con, String userId, String deviceName)
+            throws SQLException, StorageQueryException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ? AND device_name = ?;";
+
+        return update(con, QUERY, pst -> {
+            pst.setString(1, userId);
+            pst.setString(2, deviceName);
+        });
+    }
+
+    public static int removeUser_Transaction(Start start, Connection con, String userId)
+            throws SQLException, StorageQueryException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUsersTable()
+                + " WHERE user_id = ?;";
+        int removedUsersCount = update(con, QUERY, pst -> pst.setString(1, userId));
+
+        // Delete all devices and used codes for this user:
+        // This step is required only for in-memory db.
+        // Other databases will automatically delete these when the user is
+        // deleted because of foreign key constraints.
+        String QUERY2 = "DELETE FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ?;";
+        update(con, QUERY2, pst -> pst.setString(1, userId));
+
+        String QUERY3 = "DELETE FROM " + Config.getConfig(start).getTotpUsedCodesTable()
+                + " WHERE user_id = ?;";
+        update(con, QUERY3, pst -> pst.setString(1, userId));
+
+        return removedUsersCount;
+    }
+
+    public static int updateDeviceName(Start start, String userId, String oldDeviceName, String newDeviceName)
+            throws StorageQueryException, SQLException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " SET device_name = ? WHERE user_id = ? AND device_name = ?;";
+
+        return update(start, QUERY, pst -> {
+            pst.setString(1, newDeviceName);
+            pst.setString(2, userId);
+            pst.setString(3, oldDeviceName);
+        });
+    }
+
+    public static TOTPDevice[] getDevices(Start start, String userId)
+            throws StorageQueryException, SQLException {
+        String QUERY = "SELECT * FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ?;";
+
+        return execute(start, QUERY, pst -> pst.setString(1, userId), result -> {
+            List<TOTPDevice> devices = new ArrayList<>();
+            while (result.next()) {
+                devices.add(TOTPDeviceRowMapper.getInstance().map(result));
+            }
+
+            return devices.toArray(TOTPDevice[]::new);
+        });
+    }
+
+    public static int getDevicesCount_Transaction(Start start, Connection con, String userId)
+            throws StorageQueryException, SQLException {
+        String QUERY = "SELECT COUNT(*) as count FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ?;";
```

Comment:
this needs to do locking. See how other recipes do locking with in memory db. For postgres / mysql, this will be a `select ... for update` query